index.js

```diff
@@ -112,6 +122,43 @@ class Conf {
 		}
 	}
 
+	_migrate(migrations = {}, versionToMigrate = '0.0.0') {
+		const MIGRATION_KEY = '__conf-migrated-version__';
+
+		const previousMigratedVersion = this.get(MIGRATION_KEY, '0.0.0');
+
+		const olderVersionsWithMigrations = Object.keys(migrations)
+			.filter(version => this._shouldVersionPerformMigration(version, previousMigratedVersion, versionToMigrate))
+			.sort(semver.compare);
+
+		const migrationsToRun = olderVersionsWithMigrations.map(version => migrations[version]);
+
+		migrationsToRun.forEach(migration => migration(this));
+
+		this.set(MIGRATION_KEY, versionToMigrate);
+	}
+
+	/**
+	 * Specifies if a version is suitable to run its migrations.
+	 * To be suitable, a version should be comprehended between the previous migrated version and the version desired to migrate.
+	 *
+	 * @param {string} version The version that's being tested to be suitable or not.
+	 * @param {string} previousMigratedVersion The version matching the last migration that ocurred.
+	 * @param {string} versionToMigrate The version that specifies what to migrate to (a.k.a the current version).
+	 * @returns {boolean} If the version is suitable or not.
+	 */
+	_shouldVersionPerformMigration(version, previousMigratedVersion, versionToMigrate) {
```

Comment:
Nifty pick: _shouldVersionPerformMigration -> _shouldPerformMigration

Reply:
Makes perfect sense, will change 😄

Reply:
Nifty question: Should I also update the first description line of the documentation to reflect the new name?

e.g. (Specifies if a migration is suitable to run according to its version)

Reply:
I think it's fine, although I think @sindresorhus will ask you to remove the JSDoc to keep consistency since we have none atm. But I will leave that to him.

Reply:
I asked that on https://github.com/sindresorhus/conf/pull/83#discussion_r306029725

Let's just see what he says.